docs/okeps/okep-5377-extend-udn-to-support-multiple-cluster-subnets-in-layer3-topology.md

```diff
@@ -0,0 +1,124 @@
+# OKEP-5377: Extend UDN to Support Multiple Cluster Subnets in Layer3 Topology 
+
+* Issue: [#5377](https://github.com/ovn-org/ovn-kubernetes/issues/5377)
+
+## Problem Statement
+
+UDN currently supports only one cluster subnet per IP family in Layer3 topology. This limits IP
+address planning and scalability for large clusters.
+
+## Goals
+
+* Support multiple cluster subnets per IP family in UDN's Layer3 topology.
+* Align UDN’s subnet capabilities with existing support in NAD.
+* Allow dynamic subnet expansion by appending new subnets to an existing UDN.
+* Support only subnets with the same host subnet size(`hostSubnet`).
+
+## Non-Goals
+
+* Supporting multiple subnets in Layer2 or Localnet topologies.
+* Supporting deletion or shrinking of existing subnets.
+
+## Future-Goals
+
+* Support subnets with different host subnet sizes(`hostSubnet`)
```

Comment:
This is because the fix n OVN  for variable-length prefix length got merged yesterday - https://github.com/ovn-org/ovn/commit/27cc274e66acd9e0ed13525f9ea2597804107348